Sources/EmbraceCore/Embrace.swift

```diff
@@ -60,7 +60,7 @@ To start the SDK you first need to configure it using an `Embrace.Options` insta
     /// Returns the current `MetadataHandler` used to store resources and session properties.
     public let metadata: MetadataHandler
 
-    let config: EmbraceConfig
+    let config: EmbraceConfig?
```

Comment:
Would recommend we keep this as non-optional and implement this as a `protocol` to retrieve the values within the SDK. I'd prefer to have a simple object that lists out the defaults instead of having these default values live at the call site.

I'd like to avoid code like this:

```
if config.isSDKEnabled ?? false {
```

Reply:
Maybe we should provide a way to allow users to inject a config object themselves (this is the same way Android works when no appId is present). This might need some changes, in particular how we define the `EmbraceConfig`.

Reply:
Going to leave the injection of config up for a separate PR. That will require a bit of an overhaul to the structure of `EmbraceConfig`